python/GafferSceneUI/StandardOptionsUI.py

```diff
@@ -409,3 +443,59 @@ def __statisticsSummary( plug ) :
 	plugs = plugsMetadata
 
 )
+
+class _IncludedPurposesPlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	__allPurposes = [ "default", "render", "proxy", "guide" ]
+
+	def __init__( self, plugs, **kw ) :
+
+		self.__menuButton = GafferUI.MenuButton( "", menu = GafferUI.Menu( Gaffer.WeakMethod( self.__menuDefinition ) ) )
+		GafferUI.PlugValueWidget.__init__( self, self.__menuButton, plugs, **kw )
+
+		self._addPopupMenu( self.__menuButton )
+
+		self.__currentValue = None
+
+	def _updateFromValues( self, values, exception ) :
+
+		self.__currentValue = sole( values )
+		self.__menuButton.setText( ", ".join( [ p.capitalize() for p in self.__currentValue ] ) if self.__currentValue is not None else "---" )
```

Comment:
This suffers from the old "menu button draws blank when there are no values" challenge. We end up with no text or dropdown arrow if all purposes are disabled.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5470/commits/bf593aa9de398f06c19887edac0d66887bf4bc96.